src/models/ActivityModel.js

```diff
@@ -0,0 +1,279 @@
+/**
+ * @copyright Copyright (c) 2021 Louis Chemineau <louis@chmn.me>
+ *
+ * @author Louis Chemineau <louis@chmn.me>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import moment from '@nextcloud/moment'
+
+export default class ActivityModel {
+
+	_activity;
+
+	/**
+	 * Create the activity object
+	 *
+	 * @param {Object} ocsData ocs request response
+	 */
+	constructor(ocsData) {
+		if (ocsData.ocs && ocsData.ocs.data && ocsData.ocs.data[0]) {
+			ocsData = ocsData.ocs.data[0]
+		}
+
+		// convert int into boolean
+		// ocsData.hide_download = !!ocsData.hide_download
+		// ocsData.mail_send = !!ocsData.mail_send
+
+		// store state
+		this._activity = ocsData
+	}
+
+	/**
+	 * Get the activity state
+	 * ! used for reactivity purpose
+	 * Do not remove. It allow vuejs to
+	 * inject its watchers into the _activity
+	 * state and make the whole class reactive
+	 *
+	 * @returns {Object} the activity raw state
+	 * @readonly
+	 * @memberof Sidebar
+	 */
+	get state() {
+		return this._activity
+	}
```

Comment:
Should not be necessary since we switched from `#activity` to `_activity` which vue likes ;)